include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -4,15 +4,39 @@
 #include <cpp-statsd-client/UDPSender.hpp>
 #include <cstdint>
 #include <cstdio>
+#include <functional>
 #include <iomanip>
 #include <memory>
 #include <random>
 #include <sstream>
 #include <string>
+#include <utility>
 #include <vector>
 
 namespace Statsd {
 
+namespace detail {
+inline std::string sanitizePrefix(std::string prefix) {
+    // For convenience we provide the dot when generating the stat message
+    if (!prefix.empty() && prefix.back() == '.') {
+        prefix.pop_back();
+    }
+    return prefix;
+}
+
+inline float rand(unsigned int seed) {
+    static thread_local std::mt19937 twister(seed);
+    static thread_local std::uniform_real_distribution<float> dist(0.f, 1.f);
+    return dist(twister);
+}
+
+// All supported metric types
+constexpr char METRIC_TYPE_COUNT[] = "c";
+constexpr char METRIC_TYPE_GAUGE[] = "g";
+constexpr char METRIC_TYPE_TIMING[] = "ms";
+constexpr char METRIC_TYPE_SET[] = "s";
+}  // namespace detail
```

Comment:
moved this up above the constructor for the class so i could make use of the random function as a default argument